types/field_type.go

```diff
@@ -66,15 +66,38 @@ func NewFieldTypeWithCollation(tp byte, collation string, length int) *FieldType
 // Aggregation is performed by MergeFieldType function.
 func AggFieldType(tps []*FieldType) *FieldType {
 	var currType FieldType
+	isMixedSign := false
 	for i, t := range tps {
 		if i == 0 && currType.Tp == mysql.TypeUnspecified {
 			currType = *t
 			continue
 		}
 		mtp := MergeFieldType(currType.Tp, t.Tp)
+		isMixedSign = isMixedSign || (mysql.HasUnsignedFlag(currType.Flag) != mysql.HasUnsignedFlag(t.Flag))
 		currType.Tp = mtp
 		currType.Flag = mergeTypeFlag(currType.Flag, t.Flag)
 	}
+	// integral promotion when tps contains signed and unsigned
+	if isMixedSign && IsTypeInteger(currType.Tp) {
+		bumpRange := false // indicate one of tps bump currType range
+		for _, t := range tps {
+			bumpRange = bumpRange || (mysql.HasUnsignedFlag(t.Flag) && (t.Tp == currType.Tp || t.Tp == mysql.TypeBit))
```

Comment:
Why check `mysql.TypeBit` specially

Reply:
because the TypeBit can't be mergeType and assume it max range bit(64). so it must bump.
I find the all ```IntegerType``` merege  ```mysql.TypeBit```  is ```mysql.TypeVarchar```. 
It is inconsistent with mysql 8.0 <https://github.com/mysql/mysql-server/blob/8.0/sql/field.cc#L248>.

Whether it will be consistent with mysql in the future. if not, No need to consider ```mysql.TypeBit``` here